api/envoy/config/core/v3/socket_option.proto

```diff
@@ -74,6 +82,16 @@ message SocketOption {
   // The state in which the option will be applied. When used in BindConfig
   // STATE_PREBIND is currently the only valid value.
   SocketState state = 6 [(validate.rules).enum = {defined_only: true}];
+
+  // The `socket type <https://linux.die.net/man/2/socket>`_ to apply the socket option to.
+  // If not specified, the socket option will be applied to all socket types.
+  oneof type {
```

Comment:
Sorry, oneof is discouraged at the moment (interferes with incremental rollouts of new fields, see [api-style](https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md)).
Can you please change this to a message with clear order of preference?

Reply:
Done. Removed the `oneof`.

Reply:
sorry for I pointed to this wrong direction.